api_app/analyzers_manager/observable_analyzers/mnemonic_pdns.py

```diff
@@ -0,0 +1,50 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import json
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class passiveDNS(classes.ObservableAnalyzer):
+    base_url: str = "https://api.mnemonic.no/pdns/v3/"
+
+    def set_params(self, params):
+        self._cofformat = params.get("cofformat", True)
+        self._limit = params.get("limit", 100)
+
+    def run(self):
+        if self._cofformat:
+            self.base_url += "cof/"
+        try:
+            response = requests.get(
+                self.base_url + self.observable_name, data={"limit": self._limit}
+            )
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        if self._cofformat:
+            result = []
+            for i in response.text.splitlines():
+                result.append(json.loads(i))
```

Comment:
why is this logic required ? @0x0elliot 

Even in the case of `cof_format`, the response is in JSON only.

Reply:
That's because `response.json()` itself was failing normally. I had to add the responses into a list to make it JSON. The following code might help:

```py
>>> import requests
>>> r = requests.get("https://api.mnemonic.no/pdns/v3/cof/google.com")
>>> r.json()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/requests/models.py", line 888, in json
    return complexjson.loads(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/decoder.py", line 340, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 2 column 1 (char 123)
```

This is how the response for a COF response looks like:
![Screenshot 2021-12-07 at 2 07 50 PM](https://user-images.githubusercontent.com/60684641/144995139-4748783d-c94e-45e4-b5e4-3e83fa58e169.png)

Reply:
Alright. Makes sense. Though please just change it use a list comprehension:

```python3
reports = [json.loads(line) for line in response.text.splitlines()]
```